src/tests/ftest/io/ior_intercept_verify_data_integrity.py

```diff
@@ -0,0 +1,161 @@
+            write verify for 30 minutes
+
+        :avocado: tags=all,daosio,hw,full_regression,iorinterceptverifydata
+        """
+        intercept = os.path.join(self.prefix, 'lib64', 'libioil.so')
+        with_intercept = dict()
+        self.run_multiple_ior_with_pool(with_intercept, intercept)
+
+        IorCommand.log_metrics(self.log, "5 clients - with " +
+                               "interception library", with_intercept[1])
+        IorCommand.log_metrics(self.log, "1 client - without " +
+                               "interception library", with_intercept[2])
+
+    def run_multiple_ior_with_pool(self, results, intercept=None):
+        """Execute ior with optional overrides for ior flags and object_class.
+
+        If specified the ior flags and ior daos object class parameters will
+        override the values read from the yaml file.
+
+        Args:
+            intercept (str): path to the interception library. Shall be used
+                             only for POSIX through DFUSE.
+            ior_flags (str, optional): ior flags. Defaults to None.
+            object_class (str, optional): daos object class. Defaults to None.
+        """
+        # Create a pool if one does not already exist
+        if self.pool is None:
+            self.create_pool()
+        # Update IOR params with the pool
+        self.ior_cmd.set_daos_params(self.server_group, self.pool)
+
+        # start dfuse if api is POSIX
+        if self.ior_cmd.api.value == "POSIX":
+            # Connect to the pool, create container and then start dfuse
+            # Uncomment below two lines once DAOS-3355 is resolved
+            # self.pool.connect()
+            # self.create_cont()
+            self.start_dfuse()
+
+        # Create two jobs and run in parallel.
+        # Job1 will have 3 client set up to use dfuse + interception
+        # library
+        # Job2 will have 1 client set up to use only dfuse.
+        job1 = self.get_new_job(self.hostlist_clients[:-1], 1,
+                                results, intercept)
+        job2 = self.get_new_job([self.hostlist_clients[-1]], 2,
+                                results, None)
+
+        job1.start()
+        job2.start()
+        job1.join()
+        job2.join()
+
+    def get_new_job(self, clients, job_num, results, intercept=None):
+        """Create a new thread for ior run
+
+        Args:
+            clients (lst): Number of clients the ior would run against.
+            job_num (int): Assigned job number
+            results (dict): A dictionary object to store the ior metrics
+            intercept (path): Path to interception library
+        """
+        hostfile = write_host_file.write_host_file(
+            clients, self.workdir, self.hostfile_clients_slots)
+        job = threading.Thread(target=self.run_multiple_ior, args=(
+            hostfile, len(clients), results, job_num, intercept))
+        return job
+
+    def run_multiple_ior(self, hostfile, num_clients,
```

Comment:
As this function is also being used in PR#1619, I would suggest to centralize this function somewhere. May be ior_test_base.py or even a separate util file for interception whatever you feel suited.

Reply:
Would fix this in a separate PR.